### DISCUSSION TITLE
Does quarto use the pandoc cli, or the pandoc library?

### DISCUSSION THREAD

**Discussion by moonpiedumplings** [quarto-dev/quarto-cli#5770] [Q&A] [answered]
### Description

I am currently trying to package an updated pandoc and quarto for nixos, but I am having trouble with it. I got pandoc cli with the 3.0 library packaged, but the package still seems to try to download the older pandoc 2.1.9, with the cli and the library all in one. But that's more packaging issues for me to be resolved. 

Before I go further, I am wondering if quarto is calling the pandoc cli, or interacting with the pandoc library directly.

**Comment by cscheid** [comment]
> Before I go further, I am wondering if quarto is calling the pandoc cli, or interacting with the pandoc library directly.

quarto calls the pandoc cli utility.

**Reply by cscheid** [reply]
> I got pandoc cli with the 3.0 library packaged, but the package still seems to try to download the older pandoc 2.1.9, with the cli and the library all in one

I don't know how that's happening to you; we attempt to download the pandoc version described here: https://github.com/quarto-dev/quarto-cli/blob/main/configuration, currently 3.1.2.

**Reply by moonpiedumplings** [reply]
Does this mean that you include pandoc in the quarto package you distribute? Or does it attempt to set this dependency for when you package quarto.

I haven't looked at it all the way through, but Nix packages quarto by taking your x86 tarball, and it has it's own system for managing dependencies.

**Reply by cscheid** [reply, marked as answer]
> Does this mean that you include pandoc in the quarto package you distribute? 

Yes. (I really recommend that you check this yourself by grabbing the tarballs..)

> I haven't looked at it all the way through, but Nix packages quarto by taking your x86 tarball, and it has it's own system for managing dependencies.

I wouldn't know - I don't know who maintains the nix package for quarto (it's not the quarto devs).

**Reply by moonpiedumplings** [reply]
Yup, sure enough, after I remove pandoc from the dependencies of the nix package, and change some stuff to make quarto use the pandoc you distribute in your tarball, it works.
